core/src/main/codegen/templates/Parser.jj

```diff
@@ -2186,18 +2186,47 @@ SqlNode TableRef3(ExprContext exprContext, boolean lateral) :
     final SqlIdentifier alias;
     final Span s;
     SqlNodeList args;
+    final List<SqlNode> tableFuncArgs;
```

Comment:
What's the difference in `List<SqlNode>` and the `SqlNodeList` declared on the next line? Feels like we should maintain consistency if possible.

Reply:
`SqlNodeList` is a `SqlNode` which implements `List<Node>` and wraps a `List<Node>` implementation.

All the productions which parse arguments use a `List<SqlNode>` presumably because the argument list itself is never a `SqlNode`, they end up being children of a `SqlCall` or similar. So there's no upside in sticking them in a SqlNode-typed wrapper.

Reply:
Sounds good can we change the one beneath then?

Reply:
The one below (`columnAliasList`) is the return type of the `ParenthesizedSimpleIdentifierList` production, which is used by multiple other rules, and in all other cases the list becomes part of the SqlNode tree, where it needs to stay as `SqlNodeList` and not `List<Node>`. It's just this production where the extra info (source token range) isn't used.

We can change the type of the variable but it seems kind of out of scope for the task?

Something that I think would improve the code though, is taking the entire argument parsing complexity out of this production and into a separate production.

Reply:
I see there are other instances where this happens. I'm honestly not a fan of it but I guess it shouldn't prevent merging this.